lib/rubocop/ast/processed_source.rb

```diff
@@ -4,6 +4,27 @@
 
 module RuboCop
   module AST
+    # A `Prism` interface's class that provides a fixed `Prism::ParseLexResult` instead of parsing.
+    #
+    # This class implements the `parse_lex` method to return a preparsed `Prism::ParseLexResult`
+    # rather than parsing the source code. It is useful in use cases like Ruby LSP,
+    # where parsed results are reused across multiple language servers, improving performance.
```

Comment:
Are you sure that wording is correct here? (between multiple language servers) I'd assume the benefit here is that the same result is reused for different operations within the lifetime of a specific instance of the language server.

Reply:
I've updated the text to focus on an objective expectation instead of providing an example for a use case that could potentially be misleading.